java/src/test/java/ai/rapids/cudf/TableTest.java

```diff
@@ -690,6 +690,215 @@ void testInnerJoinWithOnlyCommonKeys() {
     }
   }
 
+  @Test
+  void testLowerBound() {
+    boundsEmptyInput(false);
+
+    boundsEmptyValues(false);
+
+    boundsStringCategories(false);
+
+    boundsIntsSingleCol(false);
+
+    boundsLongsSingleCol(false);
+
+    boundsLongsMultiVal(false);
+
+    boundsMultiCol(false);
+
+    boundsUnequalCols1(false);
+
+    boundsUnequalCols2(false);
+
+    boundsNulls(false);
+  }
+
+  @Test
+  void testUpperBound() {
+    boundsEmptyInput(true);
+
+    boundsEmptyValues(true);
+
+    boundsStringCategories(true);
+
+    boundsIntsSingleCol(true);
+
+    boundsLongsSingleCol(true);
+
+    boundsLongsMultiVal(true);
+
+    boundsMultiCol(true);
+
+    boundsUnequalCols1(true);
+
+    boundsUnequalCols2(true);
+
+    boundsNulls(true);
+  }
+
+  private void boundsNulls(boolean isUpperBound) {
+    boolean[] descFlags = new boolean[3];
```

Comment:
I think we should have some tests that change `descFlags`

Reply:
adding now.